kie-server-parent/kie-server-controller/kie-server-controller-standalone/src/main/java/org/kie/server/controller/service/StandaloneController.java

```diff
@@ -0,0 +1,382 @@
+            logger.warn("Template storage file name is not set - System property {} is not found or is empty",
+                        FileBasedKieServerTemplateStorage.SERVER_TEMPLATE_FILE_NAME_PROP);
+            logger.warn("Controller will attempt to use {} to store server templates",
+                        FileBasedKieServerTemplateStorage.DEFAULT_SERVER_TEMPLATE_FILENAME);
+        } else {
+            templateStorage.setTemplatesLocation(templateStorageLocation);
+            logger.debug("Server templates will be stored in {}",templateStorageLocation);
+        }
+        // Override the default value for template storage (an InMemory... version)
+        super.setTemplateStorage(this.templateStorage);
+
+        // Create the instance of RuntimeManagementService that will be used, and override the default values
+        // for template storage and the server instance manager
+        runtimeManagementService = new RuntimeManagementServiceImpl();
+        runtimeManagementService.setTemplateStorage(this.templateStorage);
+        runtimeManagementService.setKieServerInstanceManager(instanceManager);
+
+        // Create the instance of SpecManagementServiceImpl that will be used, and override the default values
+        // for template storage and the server instance manager
+        specManagementService = new SpecManagementServiceImpl();
+        specManagementService.setTemplateStorage(templateStorage);
+        specManagementService.setKieServerInstanceManager(instanceManager);
+    }
+
+
+    @PUT
+    @Path("server/{id}")
+    @Consumes({MediaType.APPLICATION_XML, MediaType.APPLICATION_JSON})
+    @Produces({MediaType.APPLICATION_XML, MediaType.APPLICATION_JSON})
+    public Response connectKieServer(@Context HttpHeaders headers, @PathParam("id")String id, String serverPayload) {
+        String contentType = getContentType(headers);
+        KieServerInfo serverInfo = unmarshal(serverPayload, contentType, KieServerInfo.class);
+
+        // Create the connection and marshal the resultant KieServerSetup
+        KieServerSetup setup = super.connect(serverInfo);
+        String responseString = marshal(contentType,setup);
+
+        // Save the server instance in the InMemoryKieServerControllerStorage
+        KieServerInstanceInfo instanceInfo = new KieServerInstanceInfo(serverInfo.getLocation(),
+                                                                       KieServerStatus.UNKNOWN,serverInfo.getCapabilities());
+        Set<KieServerInstanceInfo> infoSet = new HashSet<>();
+        infoSet.add(instanceInfo);
+        KieServerInstance instance = new KieServerInstance(serverInfo.getServerId(),serverInfo.getName(),serverInfo.getVersion(),infoSet,KieServerStatus.UNKNOWN,setup);
+        kieServerControllerStorage.store(instance);
+
+        return createCorrectVariant(responseString, headers,Response.Status.CREATED);
+    }
+
+    @PUT
+    @Path("server/{id}/containers")
```

Comment:
Missing container id in path for container save.
I think that this should be same as for REST controller from KIE-WB:
[PUT] /management/servers/{id}/containers/{containerId}